Api/Controllers/CharactersController.cs

```diff
@@ -1,35 +1,56 @@
 ﻿using DomainLayer.Constants;
+using DomainLayer.Entities;
 using Microsoft.AspNetCore.Mvc;
+using RepositoryLayer.Repositories;
+using ServiceLayer.Business;
+using WebApiLayer.UserFeatures.Requests;
 
 namespace WebApiLayer.Controllers;
 
 [Route(Constants.HTTP.API_VERSION + "/gms/characters")]
 public class CharactersController : BaseController
 {
+    private readonly ICharacterServices _characterServices;
+    public CharactersController(ICharacterServices characterServices)
+    {
+        _characterServices = characterServices;
+    }
     [HttpGet]
-    public IEnumerable<string> Get()
+    public async Task<IActionResult> GetCharacters()
     {
-        return new string[] { "value1", "value2" };
+        var cList = await _characterServices.List();
+        return Ok(cList);
     }
 
     [HttpGet("{id}")]
-    public string Get(int id)
+    public async Task<IActionResult> GetCharacter(Guid id)
     {
-        return "value";
+        var character = await _characterServices.GetById(id);
+        return Ok(character);
     }
 
     [HttpPost]
-    public void Post([FromBody] string value)
+    public async Task<IActionResult> CreateCharacter([FromBody] CreateCharacterRequest character)
     {
+        var newC = new CharacterEntity();
+        Mapper.Map(character, newC);
+        await _characterServices.Create(newC);
+        return CreatedAtAction("GetCharacter", new { id = newC.Id }, newC);
     }
 
     [HttpPut("{id}")]
-    public void Put(int id, [FromBody] string value)
+    public async Task<IActionResult> UpdateCharacter(Guid id, [FromBody] UpdateCharacterRequest character)
     {
+        var newC = await _characterServices.GetById(id);
```

Comment:
Find the entity in the repo here, why not throw it when not found but wait for the service to throw it??